dev-docs/2023-11-24 versioning ax.md

```diff
@@ -0,0 +1,106 @@
+
+It is clear they have very different versions and this is where the problem starts.
+
+By SemVer's definition, the CLI did not suffer a change dramatic enought to warrant a
+breaking change, but at the same time, it is no longer the same tool users are used to.
+
+`actyx` itself did not suffer major changes either, we deprecated some platforms but
+that does not constitute a "breaking change" in interface terms; the functionality
+itself also did not change dramatically enough to warrant a breaking change — the
+entrypoint was moved from a separate binary to a unified one.
+
+There is also the issue that other components, such as the Node Manager, were built
+around the old `actyx` version and to keep everything working, we absolutely need to
+keep that version number, question is where and how.
+
+If we were going to do "self-publishing" only — i.e. as we were doing with Actions +
+Actyx Releases — we could get away with using clap/structopt's features to distinguish
+the versions of commands, but we are going to be publishing `ax` on crates.io, and that
+warrants more care when publishing, since it means that users will be consuming this
+by means of `cargo install ax`, and we need to be careful not to break their use cases.
+
+Furthermore, the core "runnable" functionality — i.e. the `node` — was part of `actyx`
+but is now part of `ax-core` which needs to be versioned as a library, as such, changes
+to the library require bumps, which may not reflect on the `node`'s version.
+
+## The Solution
+
+To that end, we have decided to publish `ax-core` with its own version, starting at
+`0.2.0` and inside it, the `node` will keep the previous `actyx` version and the CLI
+will get a new scheme to ensure that users are able to know which version of `actyx`
+the CLI has available.
+
+The new scheme consists in ensuring that the MAJOR and MINOR versions of `ax` are the
+same as the `node` version, PATCH will reflect other changes to the CLI.
+
+Following this, we now have:
+
+- `ax-core` at version `0.2.0`
+- `node` at version `2.17`
+- `ax` at version `2.17.0`
+
+From the previous list, the new publishing "rules" become:
+
+`ax-core`
+- Features
+  - will create a minor bump in `ax-core`
+- Fixes
+  - will create a patch bump in `ax-core`
+
+`node`
+- Features
+  - will create a minor bump in `node`
+  - will create a minor bump in `ax-core`
+  - will create a minor bump in `ax`
+
+- Fixes
+  - will create a patch bump to `node` (?)
+  - will create a patch bump to `ax-core`
+  - will create a patch bump to `ax` (from the dependency version update)
+
+`ax`
+- Features & Fixes
+  - will create a patch bump to `ax`
+
+## Enforcement
+
+Versions will keep being updated by the `release` tool, which will require updating
+according to this change.
+
+The `node` version will be kept in a Rust file to be managed by the `release` tool.
+
+For example:
+
+```rust
+pub const NODE_VERSION: &str = "2.17.0"
+```
```

Comment:
we might want to add a check to `ax/build.rs` that bails out if it finds that the `ax_core` dependency specifies a different NODE_VERSION than the CARGO_PKG_VERSION (modulo patches)